doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -9,184 +9,274 @@ The git repository for the project is hosted on Github at
-    1000-line PR is acceptable, while a set of new tests introduced as a
-    1000-line PR is not.
+Any addition/change to the Chapel test system should pass testing when that
+test/directory is run with ``start_test`` (and performance tests should also
+pass testing for ``start_test -performance``).
 
-.. _Test changes:
+.. _Request feedback on your changes:
 
-Test changes
-~~~~~~~~~~~~
+Request feedback on your changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-It may be helpful to read the documentation on Chapel's `Test System`_ before
-writing or modifying tests.
+.. _Push your work:
 
-* Testing your patch:
+Push your work to your feature branch
++++++++++++++++++++++++++++++++++++++
 
-  * Any addition/change to the Chapel test system should pass testing when that
-    test/directory is run with ``start_test`` (and performance tests should
-    also pass testing for ``start_test -performance``).
+Push your changes to your feature branch on GitHub will enable others to see
+your work (see `How to push`_ for command details).  Note that if you have
+already created a pull request from a feature branch, pushing your work to that
+feature branch will update the pull request.
 
-  * Changes to the Chapel implementation should not cause
-    regressions. Developers are responsible for doing a degree of testing
-    that's appropriate for their change (see guidance below) and then can rely
-    on nightly regression testing to worry about the full cross-product of
-    configurations.
+.. _Ask for feedback early:
 
-    * At a minimum, patches should pass correctness testing for the full test/
-      directory hierarchy for:
+Ask for feedback on your branch early (optional)
+++++++++++++++++++++++++++++++++++++++++++++++++
 
-      * ``CHPL_*_PLATFORM=linux64``
-      * ``CHPL_*_COMPILER=gnu``
-      * ``CHPL_COMM=none``
-      * ``CHPL_TASKS=<default>``
+Not ready to merge yours changes, but still want to see if your work is going in
+the right direction?  Feel free to ask for early feedback!  Exposing the code is
+generally done by:
 
-    * Most developers will start by focusing on a subdirectory of tests that
-      exercise the features they changed, or test/release/ as a suite of tests
-      that exercises a rich and important slice of the language.
+- Pointing someone to your feature branch on Github, or
+- Creating a `WIP PR`_ for your feature branch.  See the `Submit pull request`_
+  section below for how to do this.
 
-    * Changes that are likely to affect multi-locale executions should also be
-      tested against tests that exercise multi-locale capabilities with
-      ``CHPL_COMM=gasnet``.  A common subset is: ``test/release/``,
-      ``test/multilocale/``, and ``test/distributions/``.
+Discussion can take place in:
+- the `WIP PR`_
+- a separate Github issue
+- the chapel-developers_ mailing list
+- a private communication
+- some other strategy agreed upon by all involved parties
```

Comment:
The `-` formatting isn't making a list here. But it might be better as some sentences anyway (takes less space).